### PR TITLE
Add permissions to workflow

### PR DESCRIPTION
**Pull request by jibrang**
I am adding permissions to the workflow as described in the code scanning alert: https://github.com/github/webpack-bundlesize-compare-action/security/code-scanning/2

### CODE REVIEW

**Review by gracepark — APPROVED** [review]

File: .github/workflows/check-dist.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@
 # For our project, we generate this file through a build process from other source files.
 # We need to make sure the checked-in `index.js` actually matches what we expect it to be.
 name: Check dist/
+permissions:
+  contents: read
+  pull-requests: write
```
Comment (peterbe) [inline .github/workflows/check-dist.yml:9]: does it need this for the `upload-artifact`?

Comment (peterbe) [inline .github/workflows/check-dist.yml:9]:
```suggestion
```

I find it strange that it needs a write-permission on `pull-requests` when all it does is `upload-artifacts`. 
Can we try without first?

**Review by peterbe — CHANGES_REQUESTED** [review]
Let's try without the write permission first.

File: .github/workflows/check-dist.yml (hunk 1 above)
Comment (jibrang) [reply .github/workflows/check-dist.yml:9]: Actually the repo says that it needs write permissions:
```
This action requires the write permission for the [permissions.pull-requests scope](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idpermissions).
```
https://github.com/github/webpack-bundlesize-compare-action

Comment (jibrang) [reply .github/workflows/check-dist.yml:9]: The repo says that it needs write permissions:
```
This action requires the write permission for the [permissions.pull-requests scope](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idpermissions).
```
https://github.com/github/webpack-bundlesize-compare-action

**Review by peterbe — APPROVED** [review]
Thank you @jibrang for checking.

**Pull request merged**